### PR TITLE
Improve zope troubleshooting

### PR DESCRIPTION
**Pull request by fraibacas**

### CODE REVIEW

**Comment by fraibacas** [comment]
This PR includes two ways we can troubleshoot what zope is doing:
1) Using zencheckzopes. It can be run from any container with access to redis. To enable it run zencheckzopes enable. Once enabled, unfinished requests can be displayed running zencheckzopes show or zencheckzopes show-summary. Once the troubleshooting is done, disable it by zencheckzopes disable.
2) Pushing to opentsdb requests that take longer than a config paramenter. It is enabled by setting "zope-request-logging" in global.conf to a value greater than zero. The value represents the min duration of requests to be logged. Opentsdb UI can be used to display graphs for the response times. The metric name is zrequest.duration.
Example: global.conf.zope-request-logging 0.2 will log all requests that take more than 0.2 seconds

File: Products/ZenUtils/requestlogging/ZopeRequestLogger.py [hunk 1]
```diff
@@ -131,82 +193,142 @@ def _load_data_to_log(self, request):
...
         data['path_info'] = request.get('PATH_INFO', default='')
-        data['action_and_method'] = ''
         data['body'] = request.get('BODY', {})
         data['action_and_method'] = self._extract_action_and_method_from_body(request.get('BODY'))
-        data['client'] = request.get('REMOTE_ADDR', default='')
-        data['start_time'] = str(request._start)
         data['http_method'] = request.get('method', default='')
-        data['xff'] = request.get('X_FORWARDED_FOR', default='')
+        data['zope_id'] = zope_id
+        #data['xff'] = request.get('X_FORWARDED_FOR', default='')
```
Comment (youngklee) [inline Products/ZenUtils/requestlogging/ZopeRequestLogger.py]: Add comment why these are commented out like you did in Ln 119

File: Products/ZenUtils/requestlogging/ZopeRequestLogger.py [hunk 2]
```diff
@@ -131,82 +193,142 @@ def _load_data_to_log(self, request):
...
-
-    def _reconnect_to_redis(self):
-        now = time.time()
-        if now - self._redis_last_connection_attemp > ZopeRequestLogger.REDIS_RECONNECTION_INTERVAL:
-            ZopeRequestLogger.LOG.info("Trying to reconnect to redis")
-            self._redis_last_connection_attemp = now
-            self._redis_client = ZopeRequestLogger.create_redis_client(self.redis_url)
-            if self._redis_client:
-                ZopeRequestLogger.LOG.info("Connected to redis")
+            log.warning("Exception extracting request fingerprint {0}".format(ex))
```
Comment (youngklee) [inline Products/ZenUtils/requestlogging/ZopeRequestLogger.py]: Exception while extracting request fingerprint...

File: Products/ZenUtils/requestlogging/ZopeRequestLogger.py [hunk 3]
```diff
@@ -131,82 +193,142 @@ def _load_data_to_log(self, request):
...
+    def _is_relevant_request(self, request):
+        """ filter requests we care about for ongoing request logging """
+        user = request._data_to_log.get('user_name')
+        path = request._data_to_log.get('path_info')
+        relevant_user = user and user != str(None)
+        relevant_path = path and not path.startswith("/++resource++")
+        return relevant_user and relevant_path
+
+    def _get_duration_metric(self, request):
+        start = request._start
```
Comment (youngklee) [inline Products/ZenUtils/requestlogging/ZopeRequestLogger.py]: unused variable 'start'

File: Products/ZenUtils/requestlogging/ZopeRequestLogger.py [hunk 4]
```diff
@@ -131,82 +193,142 @@ def _load_data_to_log(self, request):
...
             ident = str(threading.current_thread().ident)
             fingerprint = ":".join((
                 ZopeRequestLogger.REDIS_KEY_PATTERN,
+                str(random.randint(0,100)),
                 os.environ.get("CONTROLPLANE_SERVICED_ID", "X"),
-                os.environ.get("CONTROLPLANE_INSTANCE_ID", "X"),
                 str(os.getpid()),
                 ident,
+                str(zope_id),
+                str(int(request._start*1000))
```
Comment (youngklee) [inline Products/ZenUtils/requestlogging/ZopeRequestLogger.py]: Add comment why 1000 is multiplied

**Pull request merged**